salt/sdb/couchdb.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+'''
+CouchDB sdb Module
+
+:maintainer:    SaltStack
+:maturity:      New
+:depends:       python2-couchdb
+:platform:      all
+
+This allow interaction between Salt and a CouchDB [couchdb.apache.org]
+database. It uses salt's `sdb` system to allow for inserts and retrevals
+using the `sdb://` prefix in salt configuration files.
+
+To use the couchbase sdb module, it must first be configured in the salt
+master or minion config. The following arguments are required:
+
+.. code-block:: yaml
+
+    couchdb_sdb:
+      driver: couchdb
+      host: localhost
+      port: 5984
+      database: salt_sdb
+
+One could then query the CouchDB instance via an `sdb://` URI such as the
+following:
+
+.. code-block:: yaml
+
+    password: sdb://couchdb_sdb/mykey
+
+To use this interface, you must track IDs on your own or have another source
+to do the map-reduce logic necessary to calculate the ID you wish to fetch.
+
+Additional contributions to build true map-reduce functionality into this module
+would be welcome.
+'''
+
+from __future__ import absolute_import
+
+# Import Python libraries
+import logging
+from uuid import uuid4
+try:
+    import couchdb
+    HAS_COUCH = True
+except ImportError:
+    HAS_COUCH = False
+
+# Import Salt libraries
+from salt.utils.decorators import memoize
+
+log = logging.getLogger(__name__)
+
+# 'set' is a reserved word
+__func_alias__ = {'set_': 'set'}
+
+
+def __virtual__():
+    '''
+    Require the python2-couchdb libraries
+    '''
+    return HAS_COUCH
+
+
+@memoize
+def _construct_uri(profile):
+    '''
+    Examine configuration and return
+    a uri for the couchdb server in the following format:
+
+    .. code-block:: bash
+
+        http://localhost:5984/
+    '''
+    return 'http://{host}:{port}'.format(host=profile['host'], port=profile['port'])
```

Comment:
Would `'http://{host}:{port}'.format(**profile)` be okay (especially with Pedro)?